fixes/consolidate-multiple-richtext-history.sql

```diff
@@ -0,0 +1,157 @@
+-- Asset Rows
+declare @q1 varchar(max);
+select @q1 = ';
+with H as (
+	select AA.ID, AA.AssetType, AuditID, R=ROW_NUMBER() OVER(partition by AA.ID, AA.AssetType order by AuditID)
+	from dbo.AssetAudit AA
+)
+insert #assets(ID, CurrentAuditID, NextAuditID)
+select aH.ID, aH.AuditID, bH.AuditID
+from H as aH
+join dbo.' + @Table + ' aAsset on aAsset.ID=aH.ID and aAsset.AssetType=aH.AssetType and aH.AuditID = aAsset.AuditBegin
+join H as bH on bH.ID=aH.ID and aH.AssetType=bH.AssetType and bH.R=aH.R+1
+join dbo.' + @Table + ' bAsset on bAsset.ID=bH.ID and bAsset.AssetType=bH.AssetType and bH.AuditID = bAsset.AuditBegin'
+
+-- Consecutive Asset changes
+declare @q2 varchar(max);
+select @q2 = ';
+insert #suspect(ID, CurrentAuditID, NextAuditID)
+select Assets.ID, Assets.CurrentAuditID, Assets.NextAuditID
+from #assets as Assets
+join dbo.' + @Table + ' currentAsset on currentAsset.ID=Assets.ID and currentAsset.AuditBegin=Assets.CurrentAuditID
+join dbo.' + @Table + ' nextAsset on nextAsset.ID=Assets.ID and nextAsset.AuditBegin=Assets.NextAuditID
+join dbo.Audit currentA on currentA.ID = Assets.CurrentAuditID
+join dbo.Audit nextA on nextA.ID = Assets.NextAuditID
+WHERE
+ISNULL(currentA.ChangedByID,-1) = ISNULL(nextA.ChangedByID,-1)  -- Consecutive changes from the same user
+AND DATEDIFF(mi,currentA.ChangeDateUTC,nextA.ChangeDateUTC) <= ' + CAST(@timethreshold as varchar(10)) + ' --Time threshold / period / lapse.
+AND	ISNULL(currentAsset.' + @Field + ',-1) != ISNULL(nextAsset.' + @Field + ',-1) --@FieldName has changed '
+
+-- Asset column comparison
+declare @colsAB varchar(max)
+select @colsAB=(
+	select REPLACE(' and (A.{col}=C.{col} or (A.{col} is null and C.{col} is null))', '{col}', quotename(COLUMN_NAME))
+	from INFORMATION_SCHEMA.COLUMNS C
+	where C.TABLE_NAME=@TableName and COLUMN_NAME not in ('ID','AssetType','AuditBegin', @FieldName, 'AuditEnd')
+	for xml path('')
+)
+
+-- consecutive redundant entries
+declare @q3 varchar(max); 
+select @q3 = '
+insert #bad(ID, CurrentAuditID, NextAuditID)
+select _.ID, CurrentAuditID, NextAuditID
+from #suspect _
+join dbo.' + @Table + ' A on A.ID=_.ID and A.AuditBegin=_.CurrentAuditID
+join dbo.' + @Table + ' B on B.ID=_.ID and B.AuditBegin=_.NextAuditID
+join dbo.' + @Table + ' C on C.ID=_.ID and C.AuditEnd=_.CurrentAuditID AND ISNULL(C.' + @Field + ',-1) != ISNULL(A.' + @Field + ',-1)
+' + @colsAB
+
+-- rows to purge
+declare @q4 varchar(max);
+select @q4 = ';
+select cAsset.ID, cAsset.AssetType, cAsset.AuditBegin, a.ChangedByID, a.ChangeDateUTC
+from #bad b
+join dbo.' + @Table + ' cAsset ON cAsset.ID=b.ID and cAsset.AuditBegin=b.CurrentAuditID
+join dbo.Audit a on a.ID = b.CurrentAuditID'
+
+-- purge redundant rows
+declare @q5 varchar(max);
+select @q5 = ';
+delete dbo.' + @Table + '
+from #bad
+where ' + @Table + '.ID=#bad.ID and ' + @Table + '.AuditBegin=#bad.CurrentAuditID'
+
+declare @error int, @rowcount int
+
+exec(@q1 + @q2 + @q3 + @q4 + @q5)
+
+select @rowcount=@@ROWCOUNT, @error=@@ERROR
```

Comment:
["EXECUTE statements preserve the previous @@ROWCOUNT."](https://docs.microsoft.com/en-us/sql/t-sql/functions/rowcount-transact-sql?redirectedfrom=MSDN&view=sql-server-ver15)

Reply:
Well 💩 , so this code still lies about what it actually does

Reply:
I think the way we're using it, it should work
![image](https://user-images.githubusercontent.com/32138685/136838517-28ed3af8-0ec6-4870-859f-b1849e7cff74.png)

Reply:
That example isn't `exec`ing.

However, the cited documentation appears to be wrong (at least in SSMS), so the point is moot.